features/ui/button/button.tsx

```diff
@@ -18,3 +57,188 @@ export const Button = styled.button`
     padding: 0;
   }
 `;
+
+export const Container = styled.button<{
+  size: ButtonSize;
+  color: ButtonColor;
+  iconDisplay: IconDisplay;
+}>`
+  ${ButtonCSSReset}
+
+  box-sizing: border-box;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-radius: 8px;
+  gap: ${space(2)}; // relevant for spacing an icon if present
+  box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
+  ${textFont("sm", "medium")}
+
+  /* Dynamically set styles based on button size prop */
+  ${(props) => {
+    switch (props.size) {
+      case ButtonSize.sm:
+        return css`
+          padding: 8px 14px;
```

Comment:
Would it make sense to use `rem` unit here. At least for consistency? For border-radius and box-shadow `px` should be fine though